pkg/ruler/manager_test.go

```diff
@@ -29,41 +30,88 @@ func TestSyncRuleGroups(t *testing.T) {
 	m, err := NewDefaultMultiTenantManager(Config{RulePath: dir}, factory, nil, log.NewNopLogger(), nil)
 	require.NoError(t, err)
 
-	const user = "testUser"
+	const (
+		user1      = "testUser1"
+		user2      = "testUser2"
+		namespace1 = "ns1"
+		namespace2 = "ns2"
+	)
 
 	userRules := map[string]rulespb.RuleGroupList{
-		user: {
+		user1: {
 			&rulespb.RuleGroupDesc{
 				Name:      "group1",
-				Namespace: "ns",
+				Namespace: namespace1,
+				Interval:  30 * time.Second,
+				User:      user1,
+			},
+		},
+		user2: {
+			&rulespb.RuleGroupDesc{
+				Name:      "group2",
+				Namespace: namespace2,
 				Interval:  1 * time.Minute,
-				User:      user,
+				User:      user2,
 			},
 		},
 	}
 	m.SyncRuleGroups(context.Background(), userRules)
+	mgr1 := getManager(m, user1)
+	require.NotNil(t, mgr1)
+	test.Poll(t, 1*time.Second, true, func() interface{} {
+		return mgr1.(*mockRulesManager).running.Load()
+	})
 
-	mgr := getManager(m, user)
-	require.NotNil(t, mgr)
-
+	mgr2 := getManager(m, user2)
+	require.NotNil(t, mgr2)
 	test.Poll(t, 1*time.Second, true, func() interface{} {
-		return mgr.(*mockRulesManager).running.Load()
+		return mgr2.(*mockRulesManager).running.Load()
 	})
 
 	// Verify that user rule groups are now cached locally.
 	{
 		users, err := m.mapper.users()
 		require.NoError(t, err)
-		require.Equal(t, []string{user}, users)
+		require.Contains(t, users, user2)
+		require.Contains(t, users, user1)
+	}
+
+	// Verify that rule groups are now written to disk.
+	{
+		rulegroup1 := filepath.Join(m.mapper.Path, user1, namespace1)
+		f, error := m.mapper.FS.Open(rulegroup1)
```

Comment:
Here I am checking the written file instead of using m.GetRules() since this method is mocked in the test.